testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/authz/ClientScopePolicyTest.java

```diff
@@ -0,0 +1,218 @@
+import org.keycloak.admin.client.resource.ClientsResource;
+import org.keycloak.admin.client.resource.RealmResource;
+import org.keycloak.authorization.client.AuthorizationDeniedException;
+import org.keycloak.authorization.client.AuthzClient;
+import org.keycloak.representations.idm.ClientScopeRepresentation;
+import org.keycloak.representations.idm.RealmRepresentation;
+import org.keycloak.representations.idm.authorization.AuthorizationRequest;
+import org.keycloak.representations.idm.authorization.AuthorizationResponse;
+import org.keycloak.representations.idm.authorization.ClientScopePolicyRepresentation;
+import org.keycloak.representations.idm.authorization.PermissionRequest;
+import org.keycloak.representations.idm.authorization.ResourcePermissionRepresentation;
+import org.keycloak.representations.idm.authorization.ResourceRepresentation;
+import org.keycloak.testsuite.arquillian.annotation.AuthServerContainerExclude;
+import org.keycloak.testsuite.arquillian.annotation.AuthServerContainerExclude.AuthServer;
+import org.keycloak.testsuite.util.ClientBuilder;
+import org.keycloak.testsuite.util.ClientScopeBuilder;
+import org.keycloak.testsuite.util.RealmBuilder;
+import org.keycloak.testsuite.util.UserBuilder;
+
+/**
+ * @author <a href="mailto:yoshiyuki.tabata.jy@hitachi.com">Yoshiyuki Tabata</a>
+ */
+@AuthServerContainerExclude(AuthServer.REMOTE)
+public class ClientScopePolicyTest extends AbstractAuthzTest {
+
+    @Override
+    public void addTestRealms(List<RealmRepresentation> testRealms) {
+        testRealms
+            .add(RealmBuilder.create().name("authz-test").user(UserBuilder.create().username("marta").password("password"))
+                .clientScope(ClientScopeBuilder.create().name("foo").protocol("openid-connect"))
+                .clientScope(ClientScopeBuilder.create().name("bar").protocol("openid-connect"))
+                .clientScope(ClientScopeBuilder.create().name("baz").protocol("openid-connect"))
+                .clientScope(ClientScopeBuilder.create().name("to-remove-a").protocol("openid-connect"))
+                .clientScope(ClientScopeBuilder.create().name("to-remove-b").protocol("openid-connect"))
+                .client(ClientBuilder.create().clientId("resource-server-test").secret("secret")
+                    .authorizationServicesEnabled(true).redirectUris("http://localhost/resource-server-test")
+                    .addOptionalClientScopes("foo", "bar", "baz").directAccessGrants())
+                .build());
+    }
+
+    @Before
+    public void configureAuthorization() throws Exception {
+        createResource("Resource A");
+        createResource("Resource B");
+
+        createClientScopePolicy("Client Scope foo Policy", "foo", "bar");
+        createClientScopePolicyAndLastOneRequired("Client Scope bar Policy", "foo", "bar");
+        createClientScopePolicy("Client Scope To Remove Policy", "to-remove-a", "to-remove-b");
```

Comment:
The test failure is cause by this line.

Could you please:

* Move `createClientScopePolicy("Client Scope To Remove Policy", "to-remove-a", "to-remove-b");` to the `testRemovePolicyWhenRemovingScope`?

Reply:
@pedroigor 
Changed.